x/incentives/keeper/grpc_query.go

```diff
@@ -269,6 +269,89 @@ func (q Querier) getGaugeFromIDJsonBytes(ctx sdk.Context, refValue []byte) ([]ty
 	return gauges, nil
 }
 
+// GaugeFilterFn is used to apply a filter on gauges
+// It must be used in query.FilterPaginate as a condition to add the gauge to the response data
+type GaugesFilterFn func(gauge *types.Gauge) bool
+
+func (q Querier) GaugesByPoolID(goCtx context.Context, req *types.QueryGaugesByPoolIDRequest) (*types.QueryGaugesByPoolIDResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	pageRes, gauges, err := q.getGaugesWithFilter(ctx, types.KeyPrefixGauges, req.GetPagination(), func(gauge *types.Gauge) bool {
+		return gauge.IsLinkedToPool(req.GetId())
+	})
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	return &types.QueryGaugesByPoolIDResponse{
+		Gauges:     gauges,
+		Pagination: pageRes,
+	}, nil
+}
+
+func (q Querier) InternalGauges(goCtx context.Context, req *types.QueryInternalGaugesRequest) (*types.QueryInternalGaugesResponse, error) {
```

Comment:
What's the reason we're not using the methods we created in gauge.go here?

Reply:
I first created internal methods in gauge.go then I figured out that you use `FilterPagination()` function in grpc functions. Hence I did not use the created methods from gauge.go but I kept them because they could be useful. Let me know if you prefer to delete them.

Reply:
Just my personal opinion, I think its better for code sanity we dont have code we dont use! :)